semgrep/semgrep/types.py

```diff
@@ -1,6 +1,31 @@
+from enum import auto
+from enum import Enum
 from typing import Any
 from typing import Mapping
 from typing import NewType
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from semgrep.rule import Rule
+    from semgrep.rule_match import RuleMatch
 
 JsonObject = Mapping[str, Any]
+
 RuleId = NewType("RuleId", str)
+
+RuleMatchMap = Mapping["Rule", Sequence["RuleMatch"]]
```

Comment:
I think a type exists in rule_match_map.py that matches this

Reply:
@nbrahms

Reply:
If it's there we should use it.